rfd/0017-supply-chain-hardening.md

```diff
@@ -0,0 +1,59 @@
+# Supply Chain Hardening
+
+To prevent supply chain attacks take control over all software distribution channels for Teleport and 
```

Comment:
I suspect this problem statement might be a bit overly broad. Also I believe there is alot more to supply chain security than this spec really covers.

For example brandon phillips I think is on atleast his second attempt at setting up an asset transparency system: https://www.transparencylog.com/ although I still have lots of concerns about this approach. So might be too immature but would be an important capability as part of any customer verification of provided assets (basically a customer can verify they've been provided the same assets as some trusted third party).

One of the frequenters at CNCF Sig Security also did a presentation about https://in-toto.io/ awhile back which is some tooling and specification targetting software supply chain, but I haven't dove in too deeply.

For docker there is also content trust that can be opted into, but I have no idea if it's widely used and definitely introduces new key handling requirements.

Reply:
Agree, this spec addresses our CI and asset hosting dependencies.
It does not cover code dependencies. The latter is largely an unsolved problem in software engineering today.
It's probably OK to focus on CI and assets for now, but the title and description should reflect that.